### ISSUE TITLE
schedule-for-stop doesn't include trips in references

### ISSUE THREAD

**Issue by dxrdxr** [OneBusAway/onebusaway-application-modules#202]
If you look at the output to **schedule-for-stop**, the _scheduleStopTime_ includes a _tripId_ but there are no _trips_ in the _references_.

**Comment by barbeau** [comment]
@dxrdxr Thanks for the report!  Did you include `includeReferences=true` as a parameter?  Could you please add the full contents of the response as an example, as well as the request URL?

**Comment by dxrdxr** [comment]
Yes, I included references and it does return agencies, routes, and stops.

The sample URL you can use is: http://api.pugetsound.onebusaway.org/api/where/schedule-for-stop/1_75403.xml?key=TEST
Not that for each scheduleStopTime there is a tripId but no trips in references.

**Comment by kurtraschke** [comment]
https://github.com/OneBusAway/onebusaway-application-modules/blob/master/onebusaway-api-core/src/main/java/org/onebusaway/api/model/transit/BeanFactoryV2.java#L712

Ordinarily the way this would work is that there would be a call to `addToReferences()` in the relevant method in `BeanFactoryV2`.  `addToReferences()` expects an OBA model object (i.e. `Route`, `Stop`, `Trip`), not an entity ID.  However, in the case of the `StopRouteDirectionScheduleBean` (which is what's getting translated into the API response here), the bean contains a `List` of either `StopTimeInstanceBean` or `FrequencyInstanceBean`, and these beans supply a string trip ID, rather than a `Trip` instance.  It is thus necessary to get an instance of the `TransitDataService` in `BeanFactoryV2`, and use that to get a `Trip` instance from the trip ID.

https://github.com/kurtraschke/onebusaway-application-modules/commit/d17ef77efc5329725102f3862d33ac27ca73b477

The commit linked above should fix the problem, at the risk of introducing some architectural ugliness—it compiles, but I haven't tested it because I can't get the OBA quickstart to run under Java 8 (almost certainly unrelated).

**Comment by barbeau** [comment]
Thanks @kurtraschke!
